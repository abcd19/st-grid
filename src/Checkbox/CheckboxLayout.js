import * as ST from './../common'
import React from 'react';
import ReactDOM from 'react-dom';
//import './Checkbox.css'

class CheckboxLayout extends React.Component{
    
    /**
    * @constructor
    * @param {type} data
    */
    constructor(props)
    {
      super(props);
      this.props = props;
      this.onClickHandle = this.onClickHandle.bind(this);
      this._val = false;
    };
      
    onClickHandle(e)
    {      
        if(this._val == true)
        {
          this._val = false;
        }else{
          this._val = true;
        }
        
        if(ST.isFunction(this.props.onChange))
        {
          this.props.onChange(this._val);
        }
    }

    render()
    {
      
      let style ={
        backgroundPosition: '30px 15px'
      };

      if(this.props.readOnly === true)
      {
        //снимаем клик
        style.opacity = '0.5';
      }else{
        //устанавливаем клик
        style.opacity = '1';
      }

      switch(this.props.val)
      {
        case true:
          this._val = true;
          style.backgroundPosition = '0px 15px';
          break;
        default:
          this._val = false;
          style.backgroundPosition = '30px 15px';
          break;
        /*case undefined:
          this._val = 'indeterminate';
          style.backgroundPosition = '15px 15px';
          break;*/
      }

      

      return (<div className="pr-checkbox-imgCont" onClick = {this.onClickHandle} style={style}></div>);
    }
};

export {CheckboxLayout}