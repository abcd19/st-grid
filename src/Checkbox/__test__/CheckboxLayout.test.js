//общие функции
import React from 'react';
import {CheckboxLayout} from './../CheckboxLayout'
import renderer from 'react-test-renderer';
test ('Checkbox tests', ()=>{
  const component = renderer.create(<CheckboxLayout />, );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
