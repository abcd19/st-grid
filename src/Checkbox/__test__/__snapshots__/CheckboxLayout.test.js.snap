// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox tests 1`] = `
<div
  className="pr-checkbox-imgCont"
  onClick={[Function]}
  style={
    Object {
      "backgroundPosition": "30px 15px",
      "opacity": "1",
    }
  }
/>
`;
